src/components/SearchResults/SearchResults.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import { useLocation } from "react-router-dom";
+import { useProducts } from "../../utils/hooks/useProducts";
+import Paginate from "../../components/Paginate/Paginate";
+import { Link } from "react-router-dom";
+import {
+  ProductsWrapper,
+  CardContainer,
+  Card,
+  CardText,
+} from "../Products/Products.styled";
+export default function SearchResults() {
+  const { search } = useLocation();
+  const searchParams = new URLSearchParams(search);
+  const q = searchParams.get("q");
+  const [currentPage, setCurrentPage] = useState(1);
+  const { data, isLoading } = useProducts(q, currentPage);
+
+  return (
+    <ProductsWrapper>
+      {!isLoading && (
+        <CardContainer>
+          {data.results.map((product, index) => {
+            const productDetail = product.data;
```

Comment:
also, the code can acess to the properties by this way:

const {name, price, category:{slug}} = product.data